packages/atomic/src/components/search/atomic-pager/atomic-pager.ts

```diff
@@ -0,0 +1,201 @@
+   * - Use a stringified SVG to display it directly.
+   */
+  @property({reflect: true, attribute: 'next-button-icon', type: String})
+  nextButtonIcon: string = ArrowRightIcon;
+
+  private radioGroupName = randomID('atomic-pager-');
+
+  public initialize() {
+    this.validateProps();
+    this.searchStatus = buildSearchStatus(this.bindings.engine);
+    this.pager = buildPager(this.bindings.engine, {
+      options: {numberOfPages: this.numberOfPages},
+    });
+    createAppLoadedListener(this.bindings.store, (isAppLoaded) => {
+      this.isAppLoaded = isAppLoaded;
+    });
+  }
+
+  @bindingGuard()
+  @errorGuard()
+  render() {
+    if (!this.pagerState?.currentPages) {
+      return html``;
+    }
+    const currentGroupName = `${this.radioGroupName}-${this.pagerState.currentPages.join('-')}`;
+    return html`${when(
+      !this.searchStatusState.hasError &&
+        this.searchStatusState.hasResults &&
+        this.isAppLoaded,
+      () =>
+        html`
+      ${renderPagerNavigation({
+        props: {
+          i18n: this.bindings.i18n,
+        },
+      })(html`
+          ${renderPagerPreviousButton({
+            props: {
+              icon: this.previousButtonIcon,
+              disabled: !this.pagerState.hasPreviousPage,
+              i18n: this.bindings.i18n,
+              onClick: () => {
+                this.pager.previousPage();
+                this.focusOnFirstResultAndScrollToTop();
+              },
+            },
+          })}
+          ${renderPageButtons({
+            props: {
+              i18n: this.bindings.i18n,
+            },
+          })(
+            html`${this.pagerState.currentPages.map((pageNumber, index) =>
+              keyed(
+                `page-${pageNumber}-${index}`,
+                renderPagerPageButton({
+                  props: {
+                    isSelected: this.pager.isCurrentPage(pageNumber),
+                    ariaLabel: this.bindings.i18n.t('page-number', {
+                      pageNumber,
+                    }),
+                    onChecked: () => {
+                      this.pager.selectPage(pageNumber);
+                      this.focusOnFirstResultAndScrollToTop();
+                    },
+                    page: pageNumber,
+                    groupName: currentGroupName,
+                    text: pageNumber.toLocaleString(
+                      this.bindings.i18n.language
+                    ),
+                  },
+                })
+              )
+            )}`
+          )}
+          ${renderPagerNextButton({
+            props: {
+              icon: this.nextButtonIcon,
+              disabled: !this.pagerState.hasNextPage,
+              i18n: this.bindings.i18n,
+              onClick: () => {
+                this.pager.nextPage();
+                this.focusOnFirstResultAndScrollToTop();
+              },
+            },
+          })}
+        `)}`
+    )}`;
+  }
+
+  private validateProps() {
+    new Schema({
```

Comment:
A few other things about validation (in general, not just for this component):

- Instead of just letting the bueno schema throw we should set the component's `error` property when validation fails:
```typescript
private validateProps() {
  try {
    // validate bueno schema...
  } catch(error) {
    this.error = error as Error;
    return;
  }
}
```
This way we'll render the `atomic-error` component instead of just logging an error in the console.

- We should revalidate props in the `willUpdate` lifecycle method:
```typescript
willUpdate(changedProperties: Map<string, unknown>) {
  if (changedProperties.has('numberOfPages')) {
    this.validateProps();
  }
}
```
This way we'll revalidate props if they change after initialization.